src/remote.ts

```diff
@@ -273,17 +273,30 @@ export class Remote {
       [`${authorityParts[1]}`]: agent.operating_system,
     }
 
+    // VS Code ignores the connect timeout in the SSH config and uses a default
+    // of 15 seconds, which can be too short in the case where we wait for
+    // startup scripts.  For now we hardcode a longer value.
+    const connectTimeout = 1800
```

Comment:
Btw, `1800` was my best guess at a time that should be enough for most people (30 minutes), whilst avoiding being infinite. This is because changing this setting will affect non-Coder use of the Remote SSH extension too.

If `1800` is too low, we can always bump it, too. And if users don't like us changing this, we can implement some improved conditional logic, I think.

One idea I had was to read in the setting, establish connection, and then reset the setting to the old value. But constantly touching users settings also feels wrong.

PS. Should we mention https://github.com/microsoft/vscode-remote-release/issues/8519 here?

Reply:
That makes sense!  Yeah, touching the settings like this does feel pretty janky, but I think you are right about trying to reset it after connecting.  Will add.

Also, I am not completely sure why we are directly munging it with jsonc, I will try going through the extension API since that might feel a bit less sketchy.

Good call on adding the issue link!